test/e2e/framework/util.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+)
+
+// RandomSuffix provides a random string to append to certain base name.
+func RandomSuffix() string {
+	source := []rune("abcdefghijklmnopqrstuvwxyz")
```

Comment:
this... is insufficient.  We're seeding on the current time, which means parallel tests might get the same seed.

Use kube to our advantage -- generate a random hash, try to grab a "lock" by creating a cluster-scoped object, then if we fail try again with something new.  Or use generatename on a cluster-scoped object -- those are guaranteed not to conflict.